### PR TITLE
Tiwae Select All

### PR DESCRIPTION
**Pull request by nickwmurphy**
This branch adds the ability to select and deselect all checkboxes at once to Tiwae.

**Before**:
screen shot 2017-05-20 at 8 50 58 pm

**After**:
screen shot 2017-05-22 at 10 38 11 am

### CODE REVIEW

File: tables/tiwae/src/Tiwae.jsx [hunk 1]
```diff
@@ -25,20 +25,22 @@ export default class Tiwae extends React.Component {
     styles = composeStyles(baseStyles, [defaultStyles, ...props.stylesheets]);
 
     const columns = props.columns.map(option => Object.assign(option, {
-      hidden: option.hidden || option.hidden,
+      hidden: option.def.hidden,
```
Comment (nickwmurphy) [inline tables/tiwae/src/Tiwae.jsx]: This is where the initial hidden property is being passed in. Without this, the select all control cannot respect more diverse "default columns" in the future.

File: tables/tiwae/src/Tiwae.jsx [hunk 2]
```diff
@@ -25,20 +25,22 @@ export default class Tiwae extends React.Component {
...
       expanded: false,
       ghostIndex: -1,
-      columns,
       startIndex: -1
     };
   }
 
   componentDidMount() {
     window.addEventListener('click', this.onBlur);
+    this.allChecked();
```
Comment (nickwmurphy) [inline tables/tiwae/src/Tiwae.jsx]: Is this the right place for this? 

Example: default columns have some hidden and some not hidden. The select all control needs to check that before displaying.

File: tables/tiwae/src/Tiwae.jsx [hunk 3]
```diff
@@ -138,13 +152,38 @@ export default class Tiwae extends React.Component {
...
+    const { columns } = this.state;
+    if (columns.every(element => element.hidden)) {
+      this.setState({ allChecked: false });
+    } else if (columns.every(element => !element.hidden)) {
+      this.setState({ allChecked: true });
+    } else {
+      this.setState({ allChecked: false });
+    }
+  }
+
```
Comment (nickwmurphy) [inline tables/tiwae/src/Tiwae.jsx]: This method might need a better name. It is checking to see if all columns have the same hidden property or not. If all are hidden, state.allChecked is set to false, meaning the checked prop of the select all checkbox is false (and vice versa for if all are not hidden).

File: tables/tiwae/src/Tiwae.jsx [hunk 4]
```diff
@@ -87,9 +89,20 @@ export default class Tiwae extends React.Component {
...
     this.setState({ columns, ghostIndex: -1, startIndex: -1 });
+    this.allChecked();
     this.props.onChange(columns);
   }
 
+  onSelectAll = (evt) => {
+    evt.stopPropagation();
+    const { columns } = this.state;
+    columns.forEach((column) => {
+      column.hidden = this.state.allChecked;
```
Comment (nickwmurphy) [inline tables/tiwae/src/Tiwae.jsx]: Linter does not like this line. It says "no-param-reassign."

File: tables/tiwae/src/Tiwae.jsx [hunk 5]
```diff
@@ -138,13 +152,38 @@ export default class Tiwae extends React.Component {
     this.setState({ expanded: !this.state.expanded });
   }
 
+  allChecked = () => {
```
Comment (nickwmurphy) [inline tables/tiwae/src/Tiwae.jsx]: This method might need a better name. It is checking to see if all columns have the same hidden property or not. If all are hidden, state.allChecked is set to false, meaning the checked prop of the select all checkbox is false (and vice versa for if all are not hidden).

File: tables/tiwae/src/Tiwae.jsx [hunk 6]
```diff
@@ -145,6 +163,20 @@ export default class Tiwae extends React.Component {
...
+        <input
+          checked={this.state.isAllChecked}
+          className={styles.itemCheckbox}
+          onChange={this.onSelectAll}
+          type='checkbox'
+        />
+        <div className={styles.itemLabel}>Select All</div>
+      </div>
+      <span className={styles.controlDivider}>|</span>
+      <div className={styles.reset} onClick={this.onReset} >Reset to Default</div>
```
Comment (chrisarevir) [inline tables/tiwae/src/Tiwae.jsx]: NB: Extra whitespace

**Pull request merged**